### PR TITLE
Add crl renewal

### PR DESCRIPTION
**Pull request by to-kn**
Add a solution to the Problem described in #236 
Had in running in production for ~half a year now, with no problems...
Included some basic tests - feel free, to suggest any change or anything else

### CODE REVIEW

File: manifests/server.pp [hunk 1]
```diff
@@ -497,6 +511,9 @@
   $rcvbuf                    = undef,
   $shared_ca                 = undef,
   $crl_verify                = true,
+  $crl_auto_renew            = false,
```
Comment (bastelfreak) [inline manifests/server.pp]: Can you add datatypes for the new params?

**Comment by bastelfreak** [comment]
Hi @to-kn, thanks for the PR. Can you take a look at hte inline comment and the failing spec tests?

File: manifests/server.pp (hunk 1 above)
Comment (to-kn) [reply manifests/server.pp]: Shure, if you want me to do that - i did not, since none of the params has datatypes yet...

**Comment by to-kn** [comment]
Hi @bastelfreak the tests should be finde now. (the first one failing has nothing to do with the code). anything else?

File: manifests/server.pp (hunk 1 above)
Comment (bastelfreak) [reply manifests/server.pp]: yeah, we want to have datatypes for all new params

**Comment by bastelfreak** [comment]
Can you rebase please? I fixed the travis error in https://github.com/voxpupuli/puppet-openvpn/pull/257 (maybe)

**Comment by to-kn** [comment]
looks good! (they passed...)

**Comment by bastelfreak** [comment]
@to-kn can you rebase please? Afterwards I can merge it.

File: manifests/server.pp [hunk 2]
```diff
@@ -497,6 +511,9 @@
   Optional[Integer] $rcvbuf                 = undef,
   Optional[String] $shared_ca               = undef,
   Boolean $crl_verify                       = true,
+  Boolean $crl_auto_renew                   = false,
```
Comment (bastelfreak) [inline manifests/server.pp]: Datatypes \o/

**Pull request merged**